### PR TITLE
feat(proto): added `clone_all` interfaces to all proto objects (#5658)

### PR DESCRIPTION
**Pull request by gunnar-solo**
* feat(proto): added clone_all interfaces to all proto objects
* excluded `health_check.proto` from generation process
* upgraded to solo-kit v0.24.2
* added test case to indicate difference in code generation (nil --> empty slice)

closes https://github.com/solo-io/gloo/issues/5620

* change test from FIt --> It

# Description

Please include a summary of the changes.

This bug fixes ... \ This new feature can be used to ...

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5620

### CODE REVIEW

**Review by sam-heilbron — CHANGES_REQUESTED** [review]
Let's remove the graphql.proto references

**Review by ben-taussig-solo — APPROVED** [review]
LGTM!

**Comment by sam-heilbron** [comment]
Not sure why codegen is still passing, but I see `graphql.proto.sk.md` is created. Can we remove that?

**Comment by gunnar-solo** [comment]
it's probably the case that that particular autogenerated file type isn't removed explicitly when we do our cleaning operation.  Guessing it's leftover from when graphql _did_ have a proto file

it's since been deleted
...and the builds kicked again

File: Makefile [hunk 1]
```diff
@@ -184,8 +184,12 @@ generated-code: $(OUTPUT_DIR)/.generated-code verify-enterprise-protos generate-
 # TODO(EItanya): make mockgen work for gloo
 SUBDIRS:=$(shell ls -d -- */ | grep -v vendor)
 $(OUTPUT_DIR)/.generated-code:
-	go mod tidy
```
Comment (bewebi) [inline Makefile:187]: Why are we no longer tidying?

File: projects/gloo/api/external/envoy/api/v2/core/health_check.proto [hunk 2]
```diff
@@ -20,6 +20,8 @@ import "validate/validate.proto";
 import "extproto/ext.proto";
 option (extproto.equal_all) = true;
 option (extproto.hash_all) = true;
+// Who needs this anyway
```
Comment (bewebi) [inline projects/gloo/api/external/envoy/api/v2/core/health_check.proto:23]: Could we use a more specific justification comment?

**Review by bewebi — COMMENTED** [review]
Looks good, just a couple of minor comments
Also, where does the extra `.` in the generated comments come from?

File: Makefile (hunk 1 above)
Comment (gunnar-solo) [reply Makefile:187]: largely because we no longer `tidy` on `master`

Comment (gunnar-solo) [reply Makefile:187]: mmm.  looks like it was just [moved down](https://github.com/solo-io/gloo/commit/1ac616e99cdcf8f99c313d4703bfd5813edf89be#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R187)

**Review by bewebi — APPROVED** [review]

**Review by bewebi — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**